Docs: fix pydoclint errors

BenjaminMummery opened the pull request:
# The problem

We want docstring style checking to be included in our style checking. 
However, before we can turn it on, we need to resolve all the errors that already exist!

# This PR's solution

### Resolves all currently raised errors.

Common issues and the solutions taken:
- Docstrings missing summary lines - Add a summary line
- Summary lines too long - Remove detail from the summary line to the body if possible, otherwise add noqa.
- Docstrings have raises sections but no raise statements - catch and explicitely re-raise exceptions
  
### Turns on pydocstyle and pydoclint as part of our style checking (they run as part of Ruff and Flake8) respectively.

- Flake8 configuration has been moved from Makefile to pyproject.toml.
- Ruff is now included as a dev dependency, and will run locally as part of `make style` and `make style-fix`
- Adopts google convention for docstrings.
- Rules not applied:
  - D100-107: rules requiring that docstrings are present.
  - DOC201: requires a return section in function/.method docstrings
  - DOC203: requires that docstring return sections list the return types (we leave this to the signature).
  - DOC301: requires __init__() to have a docstring.
  - DOC501: requires that functions.methods with raises statements have a `Raises` section in the docstring (Disabled for abc.py).
- additionally we disable all pydocstyle rules for tests. We leave pydoclint rules intact as these will ensure that any functional information in docstrings stays up to date with signatures.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

[ORQSDK-965]


[ORQSDK-965]: https://zapatacomputing.atlassian.net/browse/ORQSDK-965?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

BenjaminMummery committed:
style: fix docstring style errors in src/orquestra/sdk/_base/_api/_config.py

[ORQSDK-965]

BenjaminMummery committed:
style: fix docstring style errors in src/orquestra/sdk/_base/_api/_wf_run.py

[ORQSDK-965]

BenjaminMummery committed:
style: fix docstring style errors in src/orquestra/sdk/_base/_ast.py

[ORQSDK-965]

BenjaminMummery committed:
style: fix docstring style errors in src/orquestra/sdk/_base/_config.py

[ORQSDK-965]

BenjaminMummery committed:
style: fix docstring style errors in src/orquestra/sdk/_base/_db/_db.py

[ORQSDK-965]

BenjaminMummery commented on src/orquestra/sdk/_base/_config.py:
```diff
@@ -393,10 +393,7 @@ def update_config(
-            - if `config_name` was resolved to "local".
-            - if `config_name` couldn't be resolved automatically.
-        KeyError:
+        RuntimeConfigError:
```
So far as I can see, ValueError and KeyError won't actually be raised here hence their removal. Please double check this in case I've missed something!

BenjaminMummery committed:
style: fix docstring style errors in src/orquestra/sdk/secrets/_client.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/secrets/_auth.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/packaging/_versions.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/kubernetes/quantity.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_ray/_id_gen.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_ray/_dag.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_ray/_client.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/serde.py

[ORQSDK-965]

BenjaminMummery committed:
fix: keep imports consistent.

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/loader.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/cli/_ui/_prompts.py

[ORQSDK-965]

BenjaminMummery committed:
fixup! docs: fix pydoclint errors in src/orquestra/sdk/_base/loader.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/cli/_ui/_presentersw.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/cli/_repos.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/abc.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_workflow.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_traversal.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_spaces/_resolver.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_spaces/_api.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_services.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_logs/_markers.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_logs/_interfaces.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_jwt.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_in_process_runtime.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_graphs.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_git_url_utils.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_dsl.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix wording.

[ORQSDK-965]

BenjaminMummery committed:
fixup! docs: fix pydoclint errors in src/orquestra/sdk/_base/_traversal.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_driver/_client.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_base/_driver/_ce_runtime.py

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydoclint errors in src/orquestra/sdk/_ray/_ray_logs.py

[ORQSDK-965]

BenjaminMummery committed:
ci: add doclinting to style.

[ORQSDK-965]

BenjaminMummery committed:
docs: fix pydocstyle errors.

[ORQSDK-965]

BenjaminMummery committed:
Merge branch 'main' into ORQSDK-965-fix-pydoclint-errors

BenjaminMummery committed:
Merge branch 'main' into ORQSDK-965-fix-pydoclint-errors

BenjaminMummery committed:
style: fix style issues left over from merge.

[ORQSDK-965]

BenjaminMummery committed:
ci: disable pydocstyle checking.

[ORQSDK-965]

BenjaminMummery committed:
docs: add explanation for DOC203

[ORQSDK-965]

BenjaminMummery committed:
docs: pass for consistent formatting.

[ORQSDK-965]

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -259,13 +270,19 @@ def create_workflow_def(
-            orquestra.sdk._base._driver._exceptions.UnknownHTTPError
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
I couldn't immediately see a good description of when this is raised, any suggestions?

BenjaminMummery replied:
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery committed:
[ORQSDK-965]

BenjaminMummery committed:
Docs: Apply pydocstyle rules.

[ORQSDK-965]

BenjaminMummery committed:
Docs: Apply pydocstyle rules.

[ORQSDK-965]

BenjaminMummery committed:
ci: add-ignores for pydocstyle

[ORQSDK-965]

BenjaminMummery committed:
Revert "ci: add-ignores for pydocstyle"

[ORQSDK-965]
This reverts commit 0cfca9b3ca3eedddf923f6f3bdddbb021e88b55e.

BenjaminMummery committed:
test: un-mock CalledProcessError.

[ORQSDK-965]

BenjaminMummery committed:
ci: manage doc rules.

[ORQSDK-965]

BenjaminMummery committed:
ci: add noqas for docstring rules.

[ORQSDK-965]

BenjaminMummery committed:
docs: update changelog.

[ORQSDK-965]

alexjuda commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -294,33 +318,56 @@ def list_workflow_defs(
         """First step in the auth flow. Fetches the URL that the user has to visit.
 
+        Args:
+            redirect_port: TODO
```
**blocking**: hanging TODO without a JIRA ticket

alexjuda commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -294,33 +318,56 @@ def list_workflow_defs(
+            KeyError: when the URL couldn't be found in the response.
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
blocking: hanging TODO without a JIRA ticket

BenjaminMummery replied:
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

alexjuda commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -374,15 +441,21 @@ def create_workflow_run(
-            orquestra.sdk._base._driver._exceptions.InvalidTokenError
-            orquestra.sdk._base._driver._exceptions.ForbiddenError
-            orquestra.sdk._base._driver._exceptions.UnknownHTTPError
+            orquestra.sdk._base._driver._exceptions.InvalidWorkflowRunRequest: TODO
```
**blocking**: hanging TODOs

alexjuda commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -459,21 +555,37 @@ def list_workflow_run_summaries(
-            orquestra.sdk._base._driver._exceptions.UnknownHTTPError
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
**blocking**: hanging TODO

BenjaminMummery replied:
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

alexjuda commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -500,13 +612,29 @@ def _list_workflow_runs(
+        Raises:
+          orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
**blocking**: hanging TODO

BenjaminMummery replied:
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

alexjuda commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -559,20 +706,21 @@ def get_workflow_run(self, wf_run_id: _models.WorkflowRunID) -> WorkflowRun:
+                workflow run cannot be found
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
**blocking**: hanging TODO

alexjuda commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -584,24 +732,36 @@ def terminate_workflow_run(
+                workflow run cannot be found
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
**blocking**: hanging TODO

BenjaminMummery replied:
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

alexjuda commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -627,17 +794,22 @@ def get_workflow_run_artifacts(
+                the artifact cannot be found.
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
**blocking**: hanging TODO

BenjaminMummery replied:
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

alexjuda commented on src/orquestra/sdk/_base/_dsl.py:
```diff
@@ -113,10 +113,9 @@ def __iter__(self):
+    """A task import that uses a private Git repo.
 
-    Please use a helper method, such as GithubImport, instead of this class
+    Please use a helper method, such as GithubImport, instead of this class.
```
**nitpick**: `GithubImport` is not a method.

```suggestion
    Please use a helper, such as GithubImport, instead of this class.
```

alexjuda commented on src/orquestra/sdk/_base/_dsl.py:
```diff
@@ -953,9 +962,14 @@ def with_custom_image(
+        """Check if an ArtifactFuture has been destructured.
+
+        Args:
+            fn_name: TODO
```
**blocking**: trailing TODOs

alexjuda commented on src/orquestra/sdk/_base/_graphs.py:
```diff
@@ -41,8 +41,11 @@ def topological_sort(graph_to_sort: Graph) -> t.List[Node]:
+            visited before any of the successors.
+
+    Raises:
+        ValueError: TODO
```
**blocking**: trailing TODO

alexjuda commented on src/orquestra/sdk/_base/_in_process_runtime.py:
```diff
@@ -112,9 +105,9 @@ def _get_kwargs(
-    time `create_workflow_run` is called.
+    """Result of calling workflow function directly.
+
+    Empty at first, filled each time `create_workflow_run` is called.
```
```suggestion
    Empty at first, filled each time ``create_workflow_run`` is called.
```

alexjuda commented on src/orquestra/sdk/_base/cli/_config/_list.py:
```diff
@@ -13,9 +11,7 @@
-    """
-    Encapsulates app-related logic for handling `orq login --list`.
-    """
+    """Encapsulates app-related logic for handling `orq login --list`."""
```
```suggestion
    """Encapsulates app-related logic for handling ``orq login --list``."""
```

alexjuda commented on src/orquestra/sdk/_base/cli/_ui/_click_default_group.py:
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2015, Heungsub Lee
+# Copyright (c) 2015 - 2023, Heungsub Lee
```
The changes in this file aren't done by Heungsub Lee, they're done by us.
```suggestion
# Copyright (c) 2015, Heungsub Lee
```

alexjuda commented on src/orquestra/sdk/_base/cli/_ui/_click_default_group.py:
```diff
@@ -81,8 +80,7 @@ def bar():
 class DefaultGroup(click.Group):
-    """Invokes a subcommand marked with `default=True` if any subcommand not
-    chosen.
+    """Invokes a subcommand marked with `default=True` if any subcommand not chosen.
```
```suggestion
    """Invokes a subcommand marked with ``default=True`` if any subcommand not chosen.
```

BenjaminMummery committed:
docs: fix rst formatting

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: fix rst formatting

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: fix rst formatting

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: fix rst formatting

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: fix workflow spelling

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: fix workflow spelling

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
style: add trailing newline

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: fix rst formatting

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: correct import path for ConfigNameNotFoundError

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: GitHubImport is not a method.

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
Merge branch 'main' into ORQSDK-965-fix-pydoclint-errors

BenjaminMummery committed:
docs: remove outdated raises section.

[ORQSDK-965]

BenjaminMummery committed:
docs: add explanation for ForbiddenError to docstrings.

[ORQSDK-965]

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -905,28 +1149,48 @@ def get_system_logs(self, wf_run_id: _models.WorkflowRunID) -> List[_models.SysL
+        Raises:
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -216,14 +215,19 @@ def create_workflow_def(
+            orquestra.sdk._base._driver._exceptions.InvalidWorkflowDef: TODO
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55) addresses ForbiddenError

BenjaminMummery committed:
docs: fix rst formatting

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: fix rst formatting

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -417,21 +497,37 @@ def list_workflow_runs(
-            orquestra.sdk._base._driver._exceptions.UnknownHTTPError
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -500,13 +612,29 @@ def _list_workflow_runs(
+        Raises:
+          orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -520,22 +648,34 @@ def _list_workflow_runs(
+                workflow run cannot be found
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -559,20 +706,21 @@ def get_workflow_run(self, wf_run_id: _models.WorkflowRunID) -> WorkflowRun:
+                workflow run cannot be found
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -584,24 +732,36 @@ def terminate_workflow_run(
+                workflow run cannot be found
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -627,17 +794,22 @@ def get_workflow_run_artifacts(
+                the artifact cannot be found.
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -661,17 +840,22 @@ def get_workflow_run_artifact(
+                workflow run cannot be found
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -695,17 +886,22 @@ def get_workflow_run_results(
+                workflow run result cannot be found
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -747,18 +950,24 @@ def get_workflow_run_result(
+                workflow run cannot be found
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -802,18 +1018,25 @@ def get_task_run_logs(
+                the logs exist, but cannot be decoded.
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -853,16 +1083,23 @@ def get_task_run_logs(
-            orquestra.sdk._base._driver._exceptions.UnknownHTTPError
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -946,17 +1217,22 @@ def list_projects(self, workspace_id):
-            orquestra.sdk._base._driver._exceptions.UnknownHTTPError
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -294,33 +318,56 @@ def list_workflow_defs(
+            KeyError: when the URL couldn't be found in the response.
+            orquestra.sdk._base._driver._exceptions.InvalidTokenError: when the
+                authorization token is rejected by the remote cluster.
+            orquestra.sdk._base._driver._exceptions.ForbiddenError: TODO
```
[2febed5](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/2febed57570e8e817479bb98b34f3c37cd35ba55)

BenjaminMummery committed:
docs: add decsription for ValueError in _graphs.

[ORQSDK-965]

BenjaminMummery committed:
docs: address hanging TODOs for raises sections.

[ORQSDK-965]

BenjaminMummery commented on src/orquestra/sdk/_base/_graphs.py:
```diff
@@ -41,7 +41,10 @@ def topological_sort(graph_to_sort: Graph) -> t.List[Node]:
+            visited before any of the successors.
+
+    Raises:
+        ValueError: TODO
```
[d0abc8b](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/d0abc8ba4bcfa312d4ace3270573bf6106c893dc)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -374,15 +441,21 @@ def create_workflow_run(
-            orquestra.sdk._base._driver._exceptions.ForbiddenError
-            orquestra.sdk._base._driver._exceptions.UnknownHTTPError
+            orquestra.sdk._base._driver._exceptions.InvalidWorkflowRunRequest: TODO
+            orquestra.sdk._base._driver._exceptions.UnsupportedSDKVersion: TODO
```
[d0abc8b](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/d0abc8ba4bcfa312d4ace3270573bf6106c893dc)

BenjaminMummery commented on src/orquestra/sdk/_ray/_ray_logs.py:
```diff
@@ -131,7 +128,8 @@ class DirectLogReader:
 
     def __init__(self, ray_temp: Path):
-        """
+        """TODO.
```
[d0abc8b](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/d0abc8ba4bcfa312d4ace3270573bf6106c893dc)

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -216,14 +215,19 @@ def create_workflow_def(
-            orquestra.sdk._base._driver._exceptions.InvalidTokenError
-            orquestra.sdk._base._driver._exceptions.ForbiddenError
-            orquestra.sdk._base._driver._exceptions.UnknownHTTPError
+            orquestra.sdk._base._driver._exceptions.InvalidWorkflowDef: TODO
```
[d0abc8b](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/d0abc8ba4bcfa312d4ace3270573bf6106c893dc)

BenjaminMummery committed:
style: fix pydocstyle style errors.

[ORQSDK-965]

BenjaminMummery committed:
ci: add commands to run ruff locally as part of "python" -m pymarkdown scan CHANGELOG.md

[ORQSDK-965]
"python" -m flake8 \
	src tests docs/examples
0
"python" -m black --check src tests docs/examples
"python" -m isort --check src tests docs/examples
"python" -m ruff --preview src tests docs/examples
"python" -m mypy src tests
Success: no issues found in 219 source files
This project passes style!

BenjaminMummery committed:
docs: fix rst formatting

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: revert copyright change

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: fix rst formatting

Co-authored-by: Alexander Juda <6004040+alexjuda@users.noreply.github.com>

BenjaminMummery committed:
docs: add description of redirect port.

[ORQSDK-965]

BenjaminMummery commented on src/orquestra/sdk/_base/_driver/_client.py:
```diff
@@ -294,33 +318,56 @@ def list_workflow_defs(
         """First step in the auth flow. Fetches the URL that the user has to visit.
 
+        Args:
+            redirect_port: TODO
```
[195f71e](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/316/commits/195f71ee0abe8036a11951f8f5265b8904c203b6)

BenjaminMummery committed:
docs: add descriptions of fn_name and assign_type

[ORQSDK-965]

alexjuda commented on Makefile:
```diff
@@ -112,6 +97,10 @@ black:
 isort:
 	$(PYTHON) -m isort --check src tests docs/examples
 
+.PHONY: ruff
+ruff:
+	$(PYTHON) -m ruff --preview src tests docs/examples
```
**praise**: very nice. There was at least a couple of times I wanted to run `make ruff` and couldn't :)
